pkg/virt-handler/vm.go

```diff
@@ -1004,12 +1012,19 @@ func (c *VirtualMachineController) Run(threadiness int, stopCh chan struct{}) {
 	defer c.Queue.ShutDown()
 	log.Log.Info("Starting virt-handler controller.")
 
+	err := exec.Command("libvirtd", "-d").Run()
+	if err != nil {
+		log.Log.Errorf("libvirt could not start %v", err)
+	}
+
 	// Wait for the domain cache to be synced
 	go c.domainInformer.Run(stopCh)
 	cache.WaitForCacheSync(stopCh, c.domainInformer.HasSynced)
 
 	go c.deviceManagerController.Run(stopCh)
 
+	go c.nodeLabeller.Run(threadiness, stopCh)
```

Comment:
I'm probably missing something...
I'm wondering why did we decide to run the node labeler in a loop? What work should it do besides exposing capabilities as labels on the node?
Couldn't we get away will just running the node labeler once on virt-handler start?

Reply:
It has to run in loops, because labeller can relabell nodes with new set of labels based what user specified in kubevirt CR (`obsoleteCPUModels` and `minCPUmodel`). Or when kvm is missing and admin adds it later, labeller can relabell the node. If we would not care about relabelling functionality, then it can run only once.

Reply:
> It has to run in loops, because labeller can relabell nodes with new set of labels based what user specified in kubevirt CR (`obsoleteCPUModels` and `minCPUmodel`)

That's a good point. I wasn't sure why the two additional informers were needed.

I think we can structure this to be similar to the device manager. Each virt-handler will have a node labeler goroutine and will register a callback to relabel() when the CR changes. 
like: `clusterConfig.SetConfigModifiedCallback(refreshLabeles)`
For example, see: https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-handler/device-manager/device_controller.go#L222

This was I think we wouldn't need the additional informers.
What do you think?

Reply:
> Or when kvm is missing and admin adds it later, labeller can relabell the node. If we would not care about relabelling functionality, then it can run only once.

This part is not very clear. virt-handler already has a device plugin for /dev/kvm. Do we also need a label?

Reply:
no we don't need another label. I meant it like: 
1) node is created without kvm
2) virt-handler is deployed on this node
3) labeller can't add any labels, because node is missing kvm
4) admin adds support for kvm to this node
5) labeller can relabell the node because it has now kvm.

Reply:
:+1: thanks for the explanation. I misunderstood this part.